spec/support/sequence_double.rb

```diff
@@ -0,0 +1,12 @@
+# Sequence two arbitrary mock implementations after each other. For example,
+# one can use this to sequence a return after an exception.
+#
+#   helloer = sequenced_double(double("Hello"), -> { raise "yo" }, lambda { 1 })
+#   SayHello.run(helloer)
+def sequenced_double(dbl, *actions)
```

Comment:
This method feels a little too clever to me. The `__send__` suggests we're using a private interface which we definitely shouldn't be doing.

Since this is only used in a couple of places, something like this might be simpler:

```ruby
method_results = [-> { raise Interrupt }, -> { nil }].to_enum
allow(some_object).to receive(:some_method) { method_results.next.call }
```

If this is hard to read, we could replace the `method_results` array with a custom object that provides a more readable interface, e.g.:

```ruby
method_results = StubbedMethodResult.raise(Interrupt).then_return(nil)
allow(some_object).to receive(:some_method) { method_results.next_result }
# or, if we want to get crazy with #to_proc methods
allow(some_object).to receive(:some_method, &method_results)
```